File: devel/bundle.py
import glob, os
from subprocess import call, run

base_dir = os.path.dirname(os.path.realpath(__file__))
app = os.path.basename(base_dir)

fragments = []
for p in glob.glob('src/layouts/**/*.html', recursive=True):
    c = '--in-file "' + (app + '/' + p).replace('\\', '/') + '"'
    fragments.append(c)

fragments = ' '.join(str(e) for e in fragments)

modules = []
for p in glob.glob('src/modules/**/*.html', recursive=True):
    c = '--in-file "' + (app + '/' + p).replace('\\', '/') + '"'
    modules.append(c)

modules = ' '.join(str(e) for e in modules)

cmd='polymer-bundler '\
    '--out-dir "build" '\
    '--manifest-out "build/manifest.json" '\
    '--root "../" '\
    '--shell "devel/src/kct-app.html" ' + fragments + ' ' + modules

# print(cmd)
os.system(cmd)

File: devel/bundle.js
const { 
    Bundler, 
    generateShellMergeStrategy, 
    generateSharedDepsMergeStrategy, 
    generateCountingSharedBundleUrlMapper
} = require('polymer-bundler');

const { 
    Analyzer, 
    FsUrlLoader, 
    FsUrlResolver,
    PackageUrlResolver 
} = require('polymer-analyzer');

const path = require('path');
const util = require('polymer-bundler/lib/url-utils');
const root = util.resolvePath(util.ensureTrailingSlash('../'));
const shell = 'devel/src/kct-app.html';

const urlLoader = new FsUrlLoader(root);
const urlResolver = new FsUrlResolver(root);
const moduleResolution = undefined;

const analyzer = new Analyzer({ 
    urlResolver: urlResolver, 
    urlLoader: urlLoader 
});

const bundler = new Bundler({
    analyzer: analyzer,
    excludes: [],
    inlineScripts: true,
    inlineCss: true,
    rewriteUrlsInTemplates: false,
    stripComments: true,
    urlMapper: generateCountingSharedBundleUrlMapper(analyzer.resolveUrl('bundle_'))
    //strategy: generateSharedDepsMergeStrategy(3),
    // urlMapper: generateCountingSharedBundleUrlMapper('devel/build/bundle_')
    // strategy: generateShellMergeStrategy(analyzer.resolveUrl(shell), 2)
    // strategy: generateShellMergeStrategy(shell)
});

bundler.generateManifest([
    shell
    //'devel/src/layouts/main-layout.html',
]).then(manifest => {
    bundler.bundle(manifest).then((result) => {
        for(const [url, doc] of result.documents) {
            console.log(url)
            // const out = util.resolvePath('devel/build', bundler.analyzer.urlResolver.relative(url));
            // const dir = path.dirname(out);
            // console.log(dir);
            // console.log(out);
        }
    }).catch(err => {
        console.log(err)
    })
});